neuralpp/symbolic/general_normalizer.py

```diff
@@ -245,7 +248,7 @@ def _eliminate(self, operation: AbelianOperation, index: Variable, constraint: C
         1. supports more operations (add, multiply, and, or, ...)
         2. supports multiple intervals & complicated constraints (e.g, 1 <= x <= 100, x != y)
         """
-        if isinstance(body, FunctionApplication) and body.function.value == functions.conditional:
+        if isinstance(body, FunctionApplication) and isinstance(body.function, Constant) and body.function.value == functions.conditional:
```

Comment:
Not seeing the purpose of this check. Also, can you make the attribute error more explanatory? :-)

Reply:
`value` is a property of `Constant`, if `body.function` is not a `Constant` (e.g., a `Variable`), it's a runtime error